### PR TITLE
Restore “mark as read” Buttons 

### PR DESCRIPTION
**Pull request by mallowigi**
Hi, I've restored the mark as read functionality. I am working with this extension a lot and it annoyed me that such a feature wasn't available so I made my own fork.

I'm not sure what your plans are, you're free to fork from my fork and add your own fixes.

In the meantime I will be using this fork in my everyday.

Thanks!

### CODE REVIEW

**Comment by fregante** [comment]
Thank you! We can merge this, but can you undo all the unrelated changes? Like the indentation and the changes to manifest.json

File: extension/github-notifications-preview.css [hunk 1]
```diff
@@ -1,113 +1,114 @@
...
 .NPG-container[open] + .notification-indicator {
-    z-index: 101;
+  z-index: 101;
 }
 
 /* Disable native tooltip when it has unread notifications */
 .Header .notification-indicator[data-ga-click$=":unread"]::after {
-    display: none;
+  /*display: none;*/
 }
```
Comment (fregante) [inline extension/github-notifications-preview.css:59]: If code is not required, it should be dropped, not just commented out

Comment (mallowigi) [reply extension/github-notifications-preview.css:59]: I was unsure why you put display none, so I left it out for your own consideration

**Comment by mallowigi** [comment]
I must admit that I didn't test Firefox. Can you do it ?

**Comment by fregante** [comment]
I don't see the buttons, but I see this, which isn't great:

File: extension/github-notifications-preview.css [hunk 2]
```diff
@@ -123,16 +118,6 @@
     display: none !important;
 }
 
-.NPG-dropdown .js-notification-action:not(.notification-action-mark-archived) {
-    display: none !important;
-}
```
Comment (fregante) [inline extension/github-notifications-preview.css:128]: This change is what's showing the overlapped icons I mentioned

File: extension/github-notifications-preview.js [hunk 3]
```diff
@@ -105,6 +107,35 @@ async function openDropdown({currentTarget: indicator}) {
 		empty(container);
 		container.append(...list);
 
+		delegate('.NPG-dropdown button', 'click', async event => {
```
Comment (fregante) [inline extension/github-notifications-preview.js:110]: This is required in the next version of `delegate-it` coming in #99 

```suggestion
		delegate(document, '.NPG-dropdown button', 'click', async event => {
```

**Comment by mallowigi** [comment]
I never developed extensions for Firefox, but I'll check anyways.

**Comment by mallowigi** [comment]
By the way, this is how it looks in Chrome: 
image

I'm using the existing behavior of the on hover buttons, I think it's a minor inconvevenience. The problem is that normally the actions are overlapping the timestamp column in the notification, but in the plugin this column is hidden in compact. 

If you want to improve it, please do, but what interests me is the functionality, not the UI. And in that terms I'm okay with my fix.

**Comment by mallowigi** [comment]
image
I've improved the UI, now I've added ellipsis to the link on hover, so that the buttons don't overlap the text

**Comment by mallowigi** [comment]
I've found out that I added the styles from github dark by mistake. I fixed that, but then it means that I'll need to add the styles to the github dark repo as well.

**Comment by mallowigi** [comment]
So are you merging ?

**Comment by mallowigi** [comment]
anyone home ?

**Comment by tanmayrajani** [comment]
Thank you so much, @mallowigi! This is great! Sorry for the delay in merging :)

I noticed a minor improvement with the save/unsave, I'll get that done soon.

**Pull request merged**